### PR TITLE
Unused import

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
TODO:

- [ ] Unit tests and/or doctests in docstrings
- [ ] Tests pass locally
- [ ] Docstrings and API docs for any new/modified user-facing classes and functions
- [ ] Changes documented in docs/release.rst
- [ ] Docs build locally
- [x] GitHub Actions CI passes
- [ ] Test coverage to 100% (Codecov passes)

### CODE REVIEW

**Comment by dstansby** [comment]
Is there a ruff rule we can enable to make removing these automatic in future?

**Comment by DimitriPapadopoulos** [comment]
[`F401`](https://docs.astral.sh/ruff/rules/unused-import/url) is supposed to detect such issues, and `F` is part of the default rulesets. However, in this specific case, ruff did not report the issue, [DeepSource](https://deepsource.com/) did.

File: numcodecs/compat.py [hunk 1]
```diff
@@ -1,9 +1,7 @@
 # flake8: noqa
```
Comment (dstansby) [inline numcodecs/compat.py:1]:
```suggestion
```
If you get rid of this line, `ruff` should run properly on this file. I tried it locally and it seems to work.

Comment (DimitriPapadopoulos) [reply numcodecs/compat.py:1]: This is handled in a different PR: #559 / https://github.com/zarr-developers/numcodecs/pull/559/commits/1c980e5f7ebb12ce9da07ed1aa08e8ff88fad5a4.

I had to silence `F401` and `E402` in preview mode.

**Pull request merged**